x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -351,8 +353,15 @@ export function getIndexPatternDatasource({
       }
```

Comment:
It seems like the `getErrorMessages` expects either an array with at least one item or `undefined` as return value. We can either handle it in here or change the consumer in workspace panel to check for empty arrays itself